milli/src/search/criteria/proximity.rs

```diff
@@ -473,14 +479,30 @@ fn resolve_plane_sweep_candidates(
             }
             Phrase(words) => {
                 let mut groups_positions = Vec::with_capacity(words.len());
-                for word in words {
-                    let positions = match words_positions.get(word) {
-                        Some(positions) => positions.iter().map(|p| (p, 0, p)).collect(),
-                        None => return Ok(vec![]),
-                    };
-                    groups_positions.push(positions);
+
+                // group stop_words together.
+                for words in words.linear_group_by_key(Option::is_none) {
+                    // skip if it's a group of stop words.
+                    if matches!(words.first(), None | Some(None)) {
+                        continue;
+                    }
+                    // make a consecutive plane-sweep on the subgroup of words.
+                    let mut subgroup = Vec::with_capacity(words.len());
+                    for word in words.into_iter().map(|w| w.as_deref().unwrap()) {
+                        match words_positions.get(word) {
+                            Some(positions) => {
+                                subgroup.push(positions.iter().map(|p| (p, 0, p)).collect())
+                            }
+                            None => return Ok(vec![]),
+                        }
+                    }
+                    groups_positions.push(plane_sweep(subgroup, true)?);
```

Comment:
A subgroup could contain only one word, isn't it an issue to call a plane-sweep with only one word?

Reply:
Oh you're right. Should I ignore such subgroups or should I push it to `groups_positions` directly instead?

Reply:
Yes!

Reply:
03eb5d87c17c545dc28aa6d886e5e6ce0befb9b7

Is this what you meant?